### PR TITLE
feat: fastify-cli, vite, and typescript

### LINKED ISSUE #366: Vitetest and Autoload
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

4.26.1

### Plugin version

5.8.0

### Node.js version

20.11.1

### Operating system

macOS

### Operating system version (i.e. 20.04, 11.3, 10)

14.2.1 (23C71)

### Description

Greetings Fellow Fastify Community!

As a long-time user, I use fastify-cli as my primary method for running my applications. The ```app.ts``` files have the typical structure for doing this with the client (after Typescript compile) with no issue, and autoload works.  I had in the past to skip doing unit testing, but this new app I am working on, unit testing is key to making sure everything is working correctly, plus if I get it working here, then others could use it as well.

In short, the unit test has this [code (view full source here)](https://github.com/JAMS-Project/node-microservice-itil/blob/1-test-build-basic-unit-test-structure/__tests__/basic.test.ts):

```ts
import fastify, {FastifyInstance } from "fastify";
import {fileURLToPath} from "node:url";
import {dirname, join} from "path";
import {describe, test, beforeEach, afterEach, expect } from 'vitest';
import buildApp from '../src/app'

const fileName = fileURLToPath(import.meta.url)
const dirName = dirname(fileName)

let server: FastifyInstance

beforeEach(async () => {
  server = await buildApp(fastify())

  await server.ready()
})
afterEach(async () => {
  // called once after all tests run
  await server.close()
})
```

However, no matter what I did, looked at, searched, etc. I keep getting this error:

```
TypeError: Unknown file extension ".ts" for /Users/hel/Documents/Shane's Personal Stuff/GIT.nosync/jams/node-microservice-itil/src/plugins/graphql.ts
    at Object.getFileProtocolModuleFormat [as file:] (node:internal/modules/esm/get_format:160:9)
    at defaultGetFormat (node:internal/modules/esm/get_format:203:36)
    at defaultLoad (node:internal/modules/esm/load:143:22)
    at ModuleLoader.load (node:internal/modules/esm/loader:409:7)
    at ModuleLoader.moduleProvider (node:internal/modules/esm/loader:291:45)
    at link (node:internal/modules/esm/module_job:76:21)
```

Now if I take out:

```ts
  server = await buildApp(fastify())
```

replace it with

```ts
  server = await fastify()
server.register(customHealthCheck)
```

... as an example... it works. However, the autoload, either directly in the unit test or loading from the function, does not work.

Many thanks in advance, as I am not sure what is going on.

### Steps to Reproduce

I have the updated code saved here with this error: https://github.com/JAMS-Project/node-microservice-itil/tree/1-test-build-basic-unit-test-structure

Updated Unit Test Here: https://github.com/JAMS-Project/node-microservice-itil/blob/1-test-build-basic-unit-test-structure/__tests__/basic.test.ts

### Expected Behavior

Unit tests should run :)

### PR DESCRIPTION
**Pull request by Bugs5382**
- major package updates
- closes #366 

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -44,31 +44,31 @@
...
     "@swc/register": "^0.1.9",
-    "@types/jest": "^29.0.0",
-    "@types/node": "^20.1.0",
-    "@types/tap": "^15.0.5",
-    "fastify": "^4.0.0-rc.2",
-    "fastify-plugin": "^4.0.0",
-    "jest": "^28.1.3",
+    "@types/jest": "^29.5.12",
+    "@types/node": "^20.11.24",
+    "@types/tap": "^15.0.11",
```
Comment (Uzlopak) [inline package.json:54]: why do we even have @types/tap? Can be removed.

**Review by Uzlopak — COMMENTED** [review]
We dont use package-lock.json. So all your "updates" for semver minor updates are not relevant. But the semver major changes are.

File: package.json (hunk 1 above)
Comment (climba03003) [reply package.json:54]: Because it is used in `typescript` testing.
https://github.com/fastify/fastify-autoload/blob/master/test/typescript/basic.ts#L1

**Comment by mcollina** [comment]
Please don't ties this to a fastify v5 release

File: package.json (hunk 1 above)
Comment (Bugs5382) [reply package.json]: Pushed new version.

@Uzlopak  and @climba03003  I am 100% ok for your updating the packages file for no package lock. I just now that there is a combination or a single package causing this issue. I am just not sure which package makes my testing work.

But this is where I have problems. I not always sure what to do in this case. In all my apps, I keep the packages  them up to date frequently for security and just keeping the code up to date. I understand that this is a plugin need to be more universal for a bigger mass. So whatever changes you think, I will accept and then test them with my code.

**Comment by mcollina** [comment]
can you please add a test?

**Comment by climba03003** [comment]
Closing due to inactivity.

**Pull request closed without merging**